### PR TITLE
feat(gastro_orders): deltas

### PR DESCRIPTION
**Pull request by alghanor**
provide initial layout including tests

https://tillhub.atlassian.net/browse/IOS-2426
https://tillhub.atlassian.net/browse/API-745

### CODE REVIEW

File: lib/v1/statuses/signatures/base.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
...F4zMC4wMDpCYXI7ODI7MTU4OzIwMjEtMDQtMDdUMTQ6NDg6MzguMDAwWjsyMDIxLTA0LTA3VDE0OjQ4OjM4LjAwMFo7ZWNkc2EtcGxhaW4tU0hBMjU2O3V0Y1RpbWU7M28rT3UwWEV2NVVFTVNnSFhPRkRNd0JOcllDYkRRSUtpZ1Nzajd6UWZWa2paYkZQUmV2WnQ4TStFUFIrdTI3eDlJY1lQS05uWkh3S2NibjBUUDl0OXc9PTtNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUVRa2VBVUFvUExnVG5VTEYyQ01PcDIxMGc2T05KWktNbHBPL3IwbDBISG40TFJIcmRjMjV4eHV0dDFLMjdXeFd2UlQyT2dVakpDdEh1TXBUb0xzdHJhZz09IiwicmV2aXNpb24iOjIsImxhdGVzdF9yZXZpc2lvbiI6MiwibG9nIjp7Im9wZXJhdGlvbiI6IkZpbmlzaFRyYW5zYWN0aW9uIiwidGltZXN0YW1wIjoxNjE3ODA2OTE4LCJ0aW1lc3RhbXBfZm9ybWF0IjoidXRjVGltZSJ9LCJzaWduYXR1cmUiOnsidmFsdWUiOiIzbytPdTBYRXY1VUVNU2dIWE9GRE13Qk5yWUNiRFFJS2lnU3NqN3pRZlZralpiRlBSZXZadDhNK0VQUit1Mjd4OUljWVBLTm5aSHdLY2JuMFRQOXQ5dz09IiwiYWxnb3JpdGhtIjoiZWNkc2EtcGxhaW4tU0hBMjU2IiwiY291bnRlciI6MTU4LCJwdWJsaWNfa2V5IjoiTUZrd0V3WUhLb1pJemowQ0FRWUlLb1pJemowREFRY0RRZ0FFUWtlQVVBb1BMZ1RuVUxGMkNNT3AyMTBnNk9OSlpLTWxwTy9yMGwwSEhuNExSSHJkYzI1eHh1dHQxSzI3V3hXdlJUMk9nVWpKQ3RIdU1wVG9Mc3RyYWc9PSJ9LCJ0c3NfaWQiOiI2Y2MyZGYzNy1kMDY1LTQ4MjQtODdmZi00OGM2ZTM3ODUwZTciLCJtZXRhZGF0YSI6e30sIl90eXBlIjoiVFJBTlNBQ1RJT04iLCJfaWQiOiI1MzQ1MTYxMy0xYTE1LTQ5MWYtYmQwMS0wMzNhNDI3MDg0YjciLCJfZW52IjoiVEVTVCIsIl92ZXJzaW9uIjoiMS4xLjAifQ=='
```
Comment (eytand) [inline lib/v1/statuses/signatures/base.js:44]: Please obfuscate the real signature.. just remove a few characters to "break" it

**Review by aleksey-a-maltsev — APPROVED** [review]

**Review by eytand — APPROVED** [review]

File: lib/v1/gastro_orders/deltas/create.request.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const base = require('./base')
+const statusCreate = require('../../statuses/create.request')
+
+module.exports = {
+  $id: 'https://schemas.tillhub.com/v1/gastro_orders.deltas.create.request.schema.json',
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'object',
+  additionalProperties: false,
+  required: [
```
Comment (leolara) [inline lib/v1/gastro_orders/deltas/create.request.js:9]: to create nothing is required?

**Comment by leolara** [comment]
what is there in gastro_orders a part from deltas?

**Comment by alghanor** [comment]
> what is there in gastro_orders a part from deltas?

as of now only deltas, we opted out of storing states in the api

**Review by leolara — APPROVED** [review]

File: lib/common/date_object.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+  type: 'object',
+  additionalProperties: false,
+  description: 'Downstream delivery of a date.',
+  required: ['amount', 'currency'],
```
Comment (boris-savic) [inline lib/common/date_object.js:5]: Should be `iso` and `unix`

File: lib/v1/gastro_orders/deltas/base.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    type: 'integer',
+    example: 34,
+    description: 'The index of this delta within the order [0,N]'
+  },
+  previous_client_id: oneOf({
+    type: 'string',
+    format: 'uuid',
+    description: 'The client id of the previous delta object - NULL for the first.'
+  }),
+  dev_uuid: {
```
Comment (boris-savic) [inline lib/v1/gastro_orders/deltas/base.js:33]: why not device then?

**Pull request merged**